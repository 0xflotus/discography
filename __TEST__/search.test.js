import testSnapshot from './fixtures/testSnapshot';

test('Should search for artist', async () => {
  const query = `
    query {
      search {
        artist (name: "Joy Orbison", first: 1) {
          edges {
            node {
              id
            }
            cursor
          }
          pageInfo {
            hasNextPage
            endCursor
          }
          totalReturned
        }
      }
    }
  `;

  await testSnapshot(query);
});

test('Should search for release', async () => {
  const query = `
    query {
      search {
        release (title: "Hyph Mngo", first: 1, after: "MTk2Njc3OQ==") {
          edges {
            node {
              id
            }
            cursor
          }
          pageInfo {
            hasNextPage
            endCursor
          }
          totalReturned
        }
      }
    }
  `;

  await testSnapshot(query);
});

test('Should search for label', async () => {
  const query = `
    query {
      search {
        label (name: "Hotflush", first: 1) {
          edges {
            node {
              id
            }
            cursor
          }
          pageInfo {
            hasNextPage
            endCursor
          }
          totalReturned
        }
      }
    }
  `;

  await testSnapshot(query);
});

test('Should search for master release', async () => {
  const query = `
    query {
      search {
        master (title: "Hyph Mngo", first: 1) {
          edges {
            node {
              id
            }
            cursor
          }
          pageInfo {
            hasNextPage
            endCursor
          }
          totalReturned
        }
      }
    }
  `;

  await testSnapshot(query);
});

test('Should apply more complex search query parameters', async () => {
  const query = `
    query {
      search {
        release(styles: "Deep House", released: "1992", country: "US", first: 3) {
          edges {
            node {
              id
              title
              styles
              released
              country
            }
          }
        }
      }
    }
  `;

  await testSnapshot(query);
});
