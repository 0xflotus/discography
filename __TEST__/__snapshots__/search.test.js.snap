// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should apply more complex search query parameters 1`] = `
Object {
  "data": Object {
    "search": Object {
      "release": Object {
        "edges": Array [
          Object {
            "node": Object {
              "country": "US",
              "id": 2151,
              "released": "1992-00-00",
              "styles": Array [
                "Deep House",
              ],
              "title": "U Got Me",
            },
          },
          Object {
            "node": Object {
              "country": "US",
              "id": 2152,
              "released": "1992-00-00",
              "styles": Array [
                "Deep House",
              ],
              "title": "Together",
            },
          },
          Object {
            "node": Object {
              "country": "US",
              "id": 2153,
              "released": "1992-00-00",
              "styles": Array [
                "Deep House",
              ],
              "title": "Some Lovin'",
            },
          },
        ],
      },
    },
  },
}
`;

exports[`Should search for artist 1`] = `
Object {
  "data": Object {
    "search": Object {
      "artist": Object {
        "edges": Array [
          Object {
            "cursor": "MTU2NDQ4Mg==",
            "node": Object {
              "id": 1564482,
            },
          },
        ],
        "pageInfo": Object {
          "endCursor": "MTU2NDQ4Mg==",
          "hasNextPage": false,
        },
        "totalReturned": 1,
      },
    },
  },
}
`;

exports[`Should search for label 1`] = `
Object {
  "data": Object {
    "search": Object {
      "label": Object {
        "edges": Array [
          Object {
            "cursor": "MjI5MTI=",
            "node": Object {
              "id": 22912,
            },
          },
        ],
        "pageInfo": Object {
          "endCursor": "MjI5MTI=",
          "hasNextPage": true,
        },
        "totalReturned": 1,
      },
    },
  },
}
`;

exports[`Should search for master release 1`] = `
Object {
  "data": Object {
    "search": Object {
      "master": Object {
        "edges": Array [
          Object {
            "cursor": "MTg4MzI1",
            "node": Object {
              "id": 188325,
            },
          },
        ],
        "pageInfo": Object {
          "endCursor": "MTg4MzI1",
          "hasNextPage": false,
        },
        "totalReturned": 1,
      },
    },
  },
}
`;

exports[`Should search for release 1`] = `
Object {
  "data": Object {
    "search": Object {
      "release": Object {
        "edges": Array [
          Object {
            "cursor": "MjA1NjUzMg==",
            "node": Object {
              "id": 2056532,
            },
          },
        ],
        "pageInfo": Object {
          "endCursor": "MjA1NjUzMg==",
          "hasNextPage": false,
        },
        "totalReturned": 1,
      },
    },
  },
}
`;
